### PR TITLE
A0-1973: Tune down contracts constants

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
# Description

We follow Alex's advice and decrease two limits for `pallet_contracts`.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- I have bumped `spec_version` and `transaction_version`

### CODE REVIEW

**Review by h4nsu — APPROVED** [review]

File: bin/runtime/src/lib.rs [hunk 1]
```diff
@@ -678,9 +678,9 @@ impl pallet_contracts::Config for Runtime {
     type DeletionQueueDepth = DeletionQueueDepth;
     type DeletionWeightLimit = DeletionWeightLimit;
     type Schedule = Schedule;
-    type CallStack = [pallet_contracts::Frame<Self>; 31];
+    type CallStack = [pallet_contracts::Frame<Self>; 5];
```
Comment (deuszx) [inline bin/runtime/src/lib.rs:681]: Just FYI - this may cause currently deployed contracts to stop working.

Comment (pmikolajczyk41) [reply bin/runtime/src/lib.rs:681]: yes, but if your contract needs 5 nested cross-contract calls, then probably you don't have an optimal architecture anyway

Comment (deuszx) [reply bin/runtime/src/lib.rs:681]: Yes. But previously you didn't know that when dry-running so could have decided to deploy it. I'm not blocking, just saying we should remember to mention it in the release notes.

Comment (DamianStraszak) [reply bin/runtime/src/lib.rs:681]: I'm kind of concerned why this is so small. Note that in EVM this is 256, so even 31 was already small, but now we are changing it to 5?! Would be great to know the core reason here...

Comment (DamianStraszak) [reply bin/runtime/src/lib.rs:681]: 5 seems too small to me, each `delegate_call`-type upgradable contract already adds 2 frames. This is severely limiting the smart contract functionality

**Comment by pmikolajczyk41** [comment]
replaced by #938

**Pull request closed without merging**